Watchman.Discord/Areas/Responses/Controllers/ResponsesController.cs

```diff
@@ -26,103 +28,94 @@ public ResponsesController(MessagesServiceFactory messagesServiceFactory, Servic
-        public async Task UpdateResponse(DiscordRequest request, Contexts contexts)
+        public async Task UpdateResponse(UpdateResponseCommand command, Contexts contexts)
         {
             var messageService = this._messagesServiceFactory.Create(contexts);
-            var onEvent = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "onevent")?.Value;
-            var message = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "message")?.Value;
-            if (onEvent == null || message == null)
+            if (command.OnEvent == null || command.Message == null)
             {
                 await messageService.SendResponse(x => x.NotEnoughArguments());
                 return;
             }
-            var response = await this._responsesService.GetResponseByOnEvent(onEvent, contexts.Server.Id);
+            var response = await this._responsesService.GetResponseByOnEvent(command.OnEvent, contexts.Server.Id);
             if (response == null)
             {
-                await messageService.SendResponse(x => x.ResponseNotFound(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseNotFound(contexts, command.OnEvent));
                 return;
             }
 
-            var defaultResponse = await this._responsesService.GetResponseByOnEvent(onEvent, DomainResponse.DEFAULT_SERVER_ID);
-            if (defaultResponse.Message == message)
+            var defaultResponse = await this._responsesService.GetResponseByOnEvent(command.OnEvent, DomainResponse.DEFAULT_SERVER_ID);
+            if (defaultResponse.Message == command.Message)
             {
                 await this._responsesService.RemoveResponse(response.OnEvent, response.ServerId);
-                await messageService.SendResponse(x => x.ResponseTheSameAsDefault(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseTheSameAsDefault(contexts, command.OnEvent));
                 return;
             }
 
-            await this._responsesService.UpdateResponse(response.Id, message);
-            await messageService.SendResponse(x => x.ResponseHasBeenUpdated(contexts, onEvent, response.Message, message));
+            await this._responsesService.UpdateResponse(response.Id, command.Message);
+            await messageService.SendResponse(x => x.ResponseHasBeenUpdated(contexts, command.OnEvent, response.Message, command.Message));
         }
 
         [AdminCommand]
-        [DiscordCommand("remove response")]
-        public async Task RemoveResponse(DiscordRequest request, Contexts contexts)
+        public async Task RemoveResponse(RemoveResponseCommand command, Contexts contexts)
         {
             var messageService = this._messagesServiceFactory.Create(contexts);
-            var onEvent = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "onevent")?.Value;
-            if (onEvent == null)
+            if (command.OnEvent == null)
```

Comment:
przecież to nigdy nie będzie nullem

Reply:
tu tak samo, ktoś kliknie enter przypadkowo zamiast podać zdarzenie i jest problem.